diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.14.0/root/etc/e-smith/events/actions/navigation-conf mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/events/actions/navigation-conf
--- e-smith-manager-1.14.0/root/etc/e-smith/events/actions/navigation-conf	2008-03-31 14:26:04.000000000 -0600
+++ mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/events/actions/navigation-conf	2008-03-31 14:26:00.000000000 -0600
@@ -81,7 +81,7 @@
 #warn "updating script $file for lang $lang\n";
 	my $navdb = $navdbs{$lang};
 	my $navinfo = NAVIGATIONDIR . "/navigation.$lang";
-	$navdb ||= esmith::ConfigDB->open($navinfo);
+	unlink $navinfo unless $navdb;
 	$navdb ||= esmith::ConfigDB->create($navinfo) or 
 	    die "Couldn't create $navinfo\n";
         $navdbs{$lang} ||= $navdb;
