--- e-smith-manager-1.14.0/root/etc/e-smith/web/functions/navigation.nav_utf8	2008-10-04 16:50:11.000000000 -0600
+++ e-smith-manager-1.14.0/root/etc/e-smith/web/functions/navigation	2008-10-04 16:51:00.000000000 -0600
@@ -30,7 +30,7 @@
 
 use esmith::cgi;
 use esmith::config;
-use esmith::ConfigDB;
+use esmith::NavigationDB;
 use esmith::util;
 use esmith::I18N;
 
@@ -185,7 +185,7 @@
 
     my $navinfo = NAVIGATIONDIR . "/navigation.$language";
 
-    my $navdb = esmith::ConfigDB->open_ro( $navinfo ) or
+    my $navdb = esmith::NavigationDB->open_ro( $navinfo ) or
 	die "Couldn't open $navinfo\n";
 
     # Check the navdb for anything with a UrlPath, which means that it doesn't
