diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.14.0/root/etc/e-smith/events/actions/navigation-conf mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/events/actions/navigation-conf
--- e-smith-manager-1.14.0/root/etc/e-smith/events/actions/navigation-conf	2008-03-26 12:25:22.000000000 -0600
+++ mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/events/actions/navigation-conf	2008-03-26 12:25:02.000000000 -0600
@@ -90,36 +90,39 @@
 
 	my $lexicon = {};
 
-	my $lfile = "/etc/e-smith/locale/$lang/etc/e-smith/web/functions/$file";
-	if (-f $lfile)
+	foreach my $lfile ( "/etc/e-smith/locale/$lang/FormMagick/general",
+			    "/etc/e-smith/locale/$lang/etc/e-smith/web/functions/$file" )
 	{
-	    # Do a quick and dirty parse of the lexicon file
-	    my $xml = $parser->parsefile($lfile);
-	    my @lexicon = @{$xml->[1]};
-	    shift @lexicon; # Remove lexicon attributes
-	    while (@lexicon)
+	    if (-f $lfile)
 	    {
-		my ($tag, $data) = splice(@lexicon, 0, 2);
-		next unless $tag eq 'entry';
-		my %entry_hash = ('attributes', @$data);
-		my $base = $entry_hash{base};
-		$base = @{$base}[2];
-		my $trans = $entry_hash{trans};
-		$trans = @{$trans}[2];
-		next unless defined $base && defined $trans;
-		$lexicon->{$base} = $trans;
+		# Do a quick and dirty parse of the lexicon file
+		my $xml = $parser->parsefile($lfile);
+		my @lexicon = @{$xml->[1]};
+		shift @lexicon; # Remove lexicon attributes
+		while (@lexicon)
+		{
+		    my ($tag, $data) = splice(@lexicon, 0, 2);
+		    next unless $tag eq 'entry';
+		    my %entry_hash = ('attributes', @$data);
+		    my $base = $entry_hash{base};
+		    $base = @{$base}[2];
+		    my $trans = $entry_hash{trans};
+		    $trans = @{$trans}[2];
+		    next unless defined $base && defined $trans;
+		    $lexicon->{$base} = $trans;
+		}
 	    }
-         }
+	}
 
-	 my $loc_heading =  localise($lexicon, $heading);
-	 $loc_heading =~ s/^\s*(\w.*?)\s*$/$1/;
-	 my $loc_description =  localise($lexicon, $description);
-	 $loc_description =~ s/^\s*(\w.*?)\s*$/$1/;
- 	 $rec->merge_props(
-	     Heading => $loc_heading,
-	     Description => $loc_description,
-	     HeadingWeight => localise($lexicon, $heading_weight),
-	     DescriptionWeight => localise($lexicon, $description_weight));
+	my $loc_heading =  localise($lexicon, $heading);
+	$loc_heading =~ s/^\s*(\w.*?)\s*$/$1/;
+	my $loc_description =  localise($lexicon, $description);
+	$loc_description =~ s/^\s*(\w.*?)\s*$/$1/;
+ 	$rec->merge_props(
+	    Heading => $loc_heading,
+	    Description => $loc_description,
+	    HeadingWeight => localise($lexicon, $heading_weight),
+	    DescriptionWeight => localise($lexicon, $description_weight));
     }
 }
 foreach my $lang (@langs)
