diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.14.0/root/etc/e-smith/web/common/cgi-bin/login mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/web/common/cgi-bin/login
--- e-smith-manager-1.14.0/root/etc/e-smith/web/common/cgi-bin/login	2007-03-12 13:29:38.000000000 -0600
+++ mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/web/common/cgi-bin/login	2007-03-12 13:27:48.000000000 -0600
@@ -38,7 +38,7 @@
 # CSS stylesheet to use (optional)
 my $STYLESHEET = '/server-common/css/tkt.css';
 # Page title (optional)
-my $TITLE = '';
+my $TITLE = 'SME Server manager';
 # For autologin, mode to fallback to if autologin fails ('login' or 'guest')
 my $AUTOLOGIN_FALLBACK_MODE = 'login';
 # Boolean flag, whether to fallback to HTTP_REFERER for back link
@@ -272,8 +272,8 @@
   }
   print <<EOD;
 <div align="center">
-<h1>$TITLE</h1>
-<p class="warning">Authorized Use Only</p>
+<p>&nbsp;</p>
+<p>&nbsp;</p>
 EOD
 
   if ($debug) {
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.14.0/root/etc/e-smith/web/common/css/tkt.css mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/web/common/css/tkt.css
--- e-smith-manager-1.14.0/root/etc/e-smith/web/common/css/tkt.css	2007-01-26 18:51:30.000000000 -0700
+++ mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/web/common/css/tkt.css	2007-03-12 13:13:56.000000000 -0600
@@ -1,6 +1,10 @@
 /* mod_auth_tkt example css */
 
-BODY {
+BODY {background-image: url(../smeserver_logo.jpg);
+	background-repeat: no-repeat;
+	background-position: 600px 40px;
+	background-position: top;
+
     font-family: arial, helvetica, sans-serif;
     font-size: small;
 }
