diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.14.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20Manager mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20Manager
--- e-smith-manager-1.14.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20Manager	2007-01-26 20:51:30.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20Manager	2007-02-13 12:46:10.000000000 -0500
@@ -14,36 +14,46 @@
 
 ErrorLog /var/log/httpd/admin_error_log
 LogLevel warn
+HERE
 
-LoadModule env_module         modules/mod_env.so
-LoadModule log_config_module  modules/mod_log_config.so
-LoadModule mime_module        modules/mod_mime.so
-LoadModule negotiation_module modules/mod_negotiation.so
-LoadModule status_module      modules/mod_status.so
-LoadModule info_module        modules/mod_info.so
-LoadModule include_module     modules/mod_include.so
-LoadModule autoindex_module   modules/mod_autoindex.so
-LoadModule dir_module         modules/mod_dir.so
-LoadModule cgi_module         modules/mod_cgi.so
-LoadModule asis_module        modules/mod_asis.so
-LoadModule imap_module        modules/mod_imap.so
-LoadModule actions_module     modules/mod_actions.so
-LoadModule userdir_module     modules/mod_userdir.so
-LoadModule proxy_module       modules/mod_proxy.so
-LoadModule proxy_http_module  modules/mod_proxy_http.so
-LoadModule alias_module       modules/mod_alias.so
-LoadModule rewrite_module     modules/mod_rewrite.so
-LoadModule access_module      modules/mod_access.so
-LoadModule auth_module        modules/mod_auth.so
-LoadModule auth_anon_module   modules/mod_auth_anon.so
-LoadModule auth_digest_module modules/mod_auth_digest.so
-LoadModule expires_module     modules/mod_expires.so
-LoadModule headers_module     modules/mod_headers.so
-LoadModule usertrack_module   modules/mod_usertrack.so
-LoadModule setenvif_module    modules/mod_setenvif.so
-
-LoadModule ssl_module        modules/mod_ssl.so
+foreach (qw(
+    env
+    log_config
+    mime
+    negotiation
+    status
+    info
+    include
+    autoindex
+    dir
+    cgi
+    asis
+    imap
+    imagemap
+    actions
+    userdir
+    proxy
+    proxy_http
+    alias
+    rewrite
+    access
+    authz_host
+    authz_user
+    auth
+    auth_anon
+    auth_digest
+    expires
+    headers
+    usertrack
+    setenvif
+    ssl
+    ))
+{
+    next unless -f "/usr/lib/httpd/modules/mod_${_}.so";
+    $OUT .= "LoadModule ${_}_module modules/mod_${_}.so\n";
+}
 
+$OUT .= <<HERE;
 PidFile /var/run/httpd-admin.pid
 ScoreBoardFile /var/run/httpd-admin.scoreboard
 UseCanonicalName off
