diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.14.0/root/etc/e-smith/events/actions/navigation-conf mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/events/actions/navigation-conf
--- e-smith-manager-1.14.0/root/etc/e-smith/events/actions/navigation-conf	2008-03-22 16:41:43.000000000 -0600
+++ mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/events/actions/navigation-conf	2008-03-22 16:41:03.000000000 -0600
@@ -44,7 +44,7 @@
 
 use XML::Parser;
 my $parser = new XML::Parser (Style => 'Tree',
-      ProtocolEncoding => 'ISO-8859-1');
+      ProtocolEncoding => 'UTF-8');
 
 foreach my $file (@files)
 {
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.14.0/root/etc/e-smith/templates/etc/e-smith/web/panels/manager/html/header.htm/10Head mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/templates/etc/e-smith/web/panels/manager/html/header.htm/10Head
--- e-smith-manager-1.14.0/root/etc/e-smith/templates/etc/e-smith/web/panels/manager/html/header.htm/10Head	2006-03-15 14:59:54.000000000 -0700
+++ mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/templates/etc/e-smith/web/panels/manager/html/header.htm/10Head	2008-03-22 16:41:16.000000000 -0600
@@ -1,7 +1,7 @@
 <html>
 <head>
 <title>Untitled Document</title>
-<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
+<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
 <link rel="stylesheet" type="text/css" href="/server-common/css/sme_core.css">
 <link rel="stylesheet" type="text/css" href="/server-common/css/sme_panel_menu.css">
 <style type="text/css">
