diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.1/root/etc/e-smith/templates/etc/e-smith/web/common/head.tmpl/30FrameSetup00 mezzanine_patched_e-smith-manager-1.13.1/root/etc/e-smith/templates/etc/e-smith/web/common/head.tmpl/30FrameSetup00
--- e-smith-manager-1.13.1/root/etc/e-smith/templates/etc/e-smith/web/common/head.tmpl/30FrameSetup00	2006-11-02 11:38:41.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.1/root/etc/e-smith/templates/etc/e-smith/web/common/head.tmpl/30FrameSetup00	2006-11-06 14:54:50.000000000 -0500
@@ -8,21 +8,6 @@
   <style type="text/css"> 
   \@import url("/server-common/css/sme_main.css");
   </style>
-  <script language="JavaScript" type="text/javascript">
-  <!-- Hide script
-  //This swap the class of the selected item.
-    function swapClass() \{
-    var i,x,tB,j=0,tA=new Array(),arg=swapClass.arguments;
-    if(document.getElementsByTagName)\{for(i=4;i<arg.length;i++)\{tB=document.getElementsByTagName(arg[i]);
-    for(x=0;x<tB.length;x++)\{tA[j]=tB[x];j++;\}\}for(i=0;i<tA.length;i++)\{
-    if(tA[i].className)\{if(tA[i].id==arg[1])\{if(arg[0]==1)\{
-    tA[i].className=(tA[i].className==arg[3])?arg[2]:arg[3];\}else\{tA[i].className=arg[2];\}
-    \}else if(arg[0]==1 && arg[1]=='none')\{if(tA[i].className==arg[2] || tA[i].className==arg[3])\{
-    tA[i].className=(tA[i].className==arg[3])?arg[2]:arg[3];\}
-    \}else if(tA[i].className==arg[2])\{tA[i].className=arg[3];\}\}\}\}\}
-    // End script hiding -->
-    </script>
-
 EOF
     \}
     $OUT;
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.1/root/etc/e-smith/web/functions/navigation mezzanine_patched_e-smith-manager-1.13.1/root/etc/e-smith/web/functions/navigation
--- e-smith-manager-1.13.1/root/etc/e-smith/web/functions/navigation	2006-11-02 11:38:41.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.1/root/etc/e-smith/web/functions/navigation	2006-11-06 14:57:14.000000000 -0500
@@ -273,6 +273,22 @@
 	print "\n<TABLE BORDER=\"0\" CELLSPACING=\"0\" CELLPADDING=\"0\">\n";
     }
 
+  print '<script language="JavaScript" type="text/javascript">
+  <!-- Hide script
+  //This swap the class of the selected item.
+    function swapClass() {
+    var i,x,tB,j=0,tA=new Array(),arg=swapClass.arguments;
+    if(document.getElementsByTagName){for(i=4;i<arg.length;i++){tB=document.getElementsByTagName(arg[i]);
+    for(x=0;x<tB.length;x++){tA[j]=tB[x];j++;}}for(i=0;i<tA.length;i++){
+    if(tA[i].className){if(tA[i].id==arg[1]){if(arg[0]==1){
+    tA[i].className=(tA[i].className==arg[3])?arg[2]:arg[3];}else{tA[i].className=arg[2];}
+    }else if(arg[0]==1 && arg[1]==\'none\'){if(tA[i].className==arg[2] || tA[i].className==arg[3]){
+    tA[i].className=(tA[i].className==arg[3])?arg[2]:arg[3];}
+    }else if(tA[i].className==arg[2]){tA[i].className=arg[3];}}}}}
+    // End script hiding -->
+    </script>
+    ';
+
     foreach $h (@sortedheadings)
     {
         if ( $NO_FRAMES )
