diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.14.0/root/etc/e-smith/web/common/cgi-bin/login mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/web/common/cgi-bin/login
--- e-smith-manager-1.14.0/root/etc/e-smith/web/common/cgi-bin/login	2007-04-09 01:45:43.000000000 -0600
+++ mezzanine_patched_e-smith-manager-1.14.0/root/etc/e-smith/web/common/cgi-bin/login	2007-04-09 01:44:23.000000000 -0600
@@ -100,6 +100,7 @@
 $back ||= $q->param($at->back_arg_name) if $at->back_arg_name;
 warn "back from cgi param is $back\n" if $back;
 $back ||= $ENV{HTTP_REFERER} if $ENV{HTTP_REFERER} && $BACK_REFERER;
+$back =~ s/^http:/https:/;
 warn "back is $back\n";
 if ($back && $back =~ m!^/!) {
   my $hostname = $server_name;
