diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/etc/e-smith/web/functions/index.cgi mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/web/functions/index.cgi
--- e-smith-manager-1.13.0/root/etc/e-smith/web/functions/index.cgi	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/web/functions/index.cgi	2006-11-01 14:41:44.000000000 -0500
@@ -0,0 +1,36 @@
+#!/usr/bin/perl -wT
+
+#----------------------------------------------------------------------
+# copyright (C) 1999-2006 Mitel Networks Corporation
+# 
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+# 		
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 		
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA
+# 
+#----------------------------------------------------------------------
+
+use strict;
+use esmith::FormMagick;
+
+my $fm = new esmith::FormMagick;
+$fm->display();
+
+exit 0;
+
+__DATA__
+<form title="FORM_TITLE" header="/etc/e-smith/web/common/noframes_head.tmpl" footer="/etc/e-smith/web/common/noframes_foot.tmpl">
+    <page name="First">
+        <description>NOFRAMES_BODY</description>
+    </page>
+ 
+</form>
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/etc/e-smith/web/functions/initial.cgi mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/web/functions/initial.cgi
--- e-smith-manager-1.13.0/root/etc/e-smith/web/functions/initial.cgi	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/web/functions/initial.cgi	2006-11-01 14:41:37.000000000 -0500
@@ -0,0 +1,36 @@
+#!/usr/bin/perl -wT
+
+#----------------------------------------------------------------------
+# copyright (C) 1999-2006 Mitel Networks Corporation
+# 
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+# 		
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 		
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA
+# 
+#----------------------------------------------------------------------
+use strict;
+use esmith::FormMagick;
+
+my $fm = new esmith::FormMagick;
+
+$fm->display();
+
+exit 0;
+
+__DATA__
+<form title="FORM_TITLE" header="/etc/e-smith/web/common/head.tmpl" footer="/etc/e-smith/web/common/foot.tmpl">
+    <page name="First">
+        <description>FRAMES_BODY</description>
+    </page>
+  
+</form>
