diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.1/createlinks mezzanine_patched_e-smith-manager-1.13.1/createlinks
--- e-smith-manager-1.13.1/createlinks	2007-01-19 07:49:53.000000000 -0700
+++ mezzanine_patched_e-smith-manager-1.13.1/createlinks	2007-01-19 07:49:17.000000000 -0700
@@ -57,8 +57,6 @@
 # Set up generic logfile timestamp renaming/symlinking
 
 foreach (qw(
-    /var/log/httpd/error_log
-    /var/log/httpd/access_log
     /var/log/httpd/admin_access_log
     /var/log/httpd/admin_error_log
     ))
