diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/createlinks mezzanine_patched_e-smith-manager-1.13.0/createlinks
--- e-smith-manager-1.13.0/createlinks	2006-11-01 21:06:06.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/createlinks	2006-11-01 21:05:46.000000000 -0500
@@ -41,3 +41,22 @@
 	logrotate
 	));
 
+# conf-httpd-admin
+
+templates2events("/etc/httpd/admin-conf/httpd.conf", qw(
+	console-save
+	bootstrap-console-save
+	network-create
+	network-delete
+	remoteaccess-update
+	logrotate
+	));
+
+# Symlink httpd-admin to httpd.
+safe_symlink("httpd", "root/usr/sbin/httpd-admin");
+
+# Set up links to daemontools.
+safe_symlink("daemontools", "root/etc/rc.d/init.d/httpd-admin");
+service_link_enhanced("httpd-admin", "K15", "6");
+service_link_enhanced("httpd-admin", "K15", "0");
+
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/access mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/access
--- e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/access	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/access	2006-11-01 20:54:22.000000000 -0500
@@ -0,0 +1 @@
+localhost
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/PermitPlainTextAccess mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/PermitPlainTextAccess
--- e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/PermitPlainTextAccess	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/PermitPlainTextAccess	2006-11-01 20:54:22.000000000 -0500
@@ -0,0 +1 @@
+no
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/status mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/status
--- e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/status	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/status	2006-11-01 20:54:22.000000000 -0500
@@ -0,0 +1 @@
+enabled
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/TCPPort mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/TCPPort
--- e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/TCPPort	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/TCPPort	2006-11-01 20:54:22.000000000 -0500
@@ -0,0 +1 @@
+980
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/type mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/type
--- e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/type	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/db/configuration/defaults/httpd-admin/type	2006-11-01 20:54:22.000000000 -0500
@@ -0,0 +1 @@
+service
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/01localAccessString mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/01localAccessString
--- e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/01localAccessString	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/01localAccessString	2006-11-01 20:52:28.000000000 -0500
@@ -0,0 +1,22 @@
+{
+    #---------------------------------------------------------------------
+    # Grab ValidFrom access list property of httpd-admin
+    # SSL enabled virtual hosts should only allow access from IP's in
+    # this list, as well as local networks.
+    #---------------------------------------------------------------------
+    use esmith::NetworksDB;
+
+    my $ndb = esmith::NetworksDB->open_ro();
+
+    my @localAccess = $ndb->local_access_spec();
+    my $validFrom = ${'httpd-admin'}{'ValidFrom'};
+    if ($validFrom)
+    {
+	push @localAccess, split /,/, $validFrom;
+    }
+    $localAccess .= join ' ',
+	map { s:/255.255.255.255::; $_ }
+	    @localAccess;
+
+    "";
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20Manager mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20Manager
--- e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20Manager	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20Manager	2006-11-01 20:52:28.000000000 -0500
@@ -0,0 +1,158 @@
+{
+    $OUT .= "Listen 127.0.0.1:${'httpd-admin'}{TCPPort}\n";
+
+    $OUT .= <<HERE;
+
+HostnameLookups off
+
+ServerAdmin admin@$DomainName
+ServerRoot /etc/httpd
+ServerTokens ProductOnly
+
+User admin
+Group admin
+
+ErrorLog /var/log/httpd/admin_error_log
+LogLevel warn
+
+LoadModule env_module         modules/mod_env.so
+LoadModule log_config_module  modules/mod_log_config.so
+LoadModule mime_module        modules/mod_mime.so
+LoadModule negotiation_module modules/mod_negotiation.so
+LoadModule status_module      modules/mod_status.so
+LoadModule info_module        modules/mod_info.so
+LoadModule include_module     modules/mod_include.so
+LoadModule autoindex_module   modules/mod_autoindex.so
+LoadModule dir_module         modules/mod_dir.so
+LoadModule cgi_module         modules/mod_cgi.so
+LoadModule asis_module        modules/mod_asis.so
+LoadModule imap_module        modules/mod_imap.so
+LoadModule actions_module     modules/mod_actions.so
+LoadModule userdir_module     modules/mod_userdir.so
+LoadModule proxy_module       modules/mod_proxy.so
+LoadModule proxy_http_module  modules/mod_proxy_http.so
+LoadModule alias_module       modules/mod_alias.so
+LoadModule rewrite_module     modules/mod_rewrite.so
+LoadModule access_module      modules/mod_access.so
+LoadModule auth_module        modules/mod_auth.so
+LoadModule auth_anon_module   modules/mod_auth_anon.so
+LoadModule auth_digest_module modules/mod_auth_digest.so
+LoadModule expires_module     modules/mod_expires.so
+LoadModule headers_module     modules/mod_headers.so
+LoadModule usertrack_module   modules/mod_usertrack.so
+LoadModule setenvif_module    modules/mod_setenvif.so
+
+LoadModule external_auth_module modules/mod_auth_external.so
+
+LoadModule ssl_module        modules/mod_ssl.so
+
+AddExternalAuth pwauth /usr/lib/httpd/modules/pwauth
+SetExternalAuthMethod pwauth pipe
+
+
+PidFile /var/run/httpd-admin.pid
+ScoreBoardFile /var/run/httpd-admin.scoreboard
+UseCanonicalName off
+LogFormat "%h %l %u %t \\"%r\\" %>s %b" common
+LogFormat "%{User-agent}i" agent
+
+CustomLog /var/log/httpd/admin_access_log common
+
+KeepAlive On
+MaxKeepAliveRequests 100
+KeepAliveTimeout 15
+
+MaxClients 150
+MaxRequestsPerChild 100
+
+ServerName www.$DomainName
+
+MinSpareServers 1
+MaxSpareServers 5
+StartServers 1
+Timeout 300
+
+DefaultIcon /icons/unknown.gif
+DirectoryIndex index.htm index.html index.shtml index.cgi
+IndexOptions FancyIndexing VersionSort NameWidth=*
+IndexIgnore .??* *~ *# HEADER* README* RCS CVS *,v *,t
+AccessFileName .htaccess
+
+AddIconByEncoding (CMP,/icons/compressed.gif) x-compress x-gzip
+AddIconByType (TXT,/icons/text.gif) text/*
+AddIconByType (IMG,/icons/image2.gif) image/*
+AddIconByType (SND,/icons/sound2.gif) audio/*
+AddIconByType (VID,/icons/movie.gif) video/*
+DefaultType text/plain
+TypesConfig /etc/mime.types
+
+AddEncoding x-compress Z
+AddEncoding x-gzip gz
+
+AddIcon /icons/binary.gif .bin .exe
+AddIcon /icons/binhex.gif .hqx
+AddIcon /icons/tar.gif .tar
+AddIcon /icons/world2.gif .wrl .wrl.gz .vrml .vrm .iv
+AddIcon /icons/compressed.gif .Z .z .tgz .gz .zip
+AddIcon /icons/a.gif .ps .ai .eps
+AddIcon /icons/layout.gif .html .shtml .htm .pdf
+AddIcon /icons/text.gif .txt
+AddIcon /icons/c.gif .c
+AddIcon /icons/p.gif .pl .py
+AddIcon /icons/f.gif .for
+AddIcon /icons/dvi.gif .dvi
+AddIcon /icons/uuencoded.gif .uu
+AddIcon /icons/script.gif .conf .sh .shar .csh .ksh .tcl
+AddIcon /icons/tex.gif .tex
+AddIcon /icons/bomb.gif core
+
+AddIcon /icons/back.gif ..
+AddIcon /icons/hand.right.gif README
+AddIcon /icons/folder.gif ^^DIRECTORY^^
+AddIcon /icons/blank.gif ^^BLANKICON^^
+
+AddLanguage en .en
+AddLanguage fr .fr
+AddLanguage de .de
+AddLanguage da .da
+AddLanguage el .el
+AddLanguage it .it
+
+LanguagePriority en fr de
+
+AddType text/html .shtml
+AddType application/x-pkcs7-crl    .crl
+
+AddType application/x-x509-ca-cert .crt
+
+BrowserMatch "Mozilla/2" nokeepalive
+BrowserMatch "MSIE 4\.0b2;" nokeepalive downgrade-1.0 force-response-1.0
+BrowserMatch "RealPlayer 4\.0" force-response-1.0
+BrowserMatch "Java/1\.0" force-response-1.0
+BrowserMatch "JDK/1\.0" force-response-1.0
+
+AddHandler cgi-script .cgi
+AddHandler server-parsed .shtml
+AddHandler imap-file map
+
+DocumentRoot         /etc/e-smith/web/panels/manager/html
+
+Alias /server-common/ /etc/e-smith/web/common/
+ScriptAlias /server-manager/noframes    /etc/e-smith/web/panels/manager/cgi-bin/noframes
+ScriptAlias /server-manager/support    /etc/e-smith/web/panels/manager/cgi-bin/support
+ScriptAlias /server-manager/navigation    /etc/e-smith/web/panels/manager/cgi-bin/navigation
+
+# e-smith manager panel
+ScriptAlias /server-manager/cgi-bin /etc/e-smith/web/panels/manager/cgi-bin
+Alias       /server-manager /etc/e-smith/web/panels/manager/html
+
+# e-smith password panel
+ScriptAlias /user-password /etc/e-smith/web/panels/password/cgi-bin/userpassword
+
+Alias /server-resources/ /home/e-smith/files/server-resources/
+
+Alias /icons/ /var/www/icons/
+
+HERE
+}
+
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/85DefaultAccess mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/85DefaultAccess
--- e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/85DefaultAccess	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/85DefaultAccess	2006-11-01 20:52:28.000000000 -0500
@@ -0,0 +1,12 @@
+
+# First, we configure the "default" to be a very restrictive set of 
+# permissions.  
+
+<Directory />
+    Options None
+    AllowOverride None
+    order deny,allow
+    deny from all
+    allow from none
+</Directory>
+
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/85ServerResourcesAccess mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/85ServerResourcesAccess
--- e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/85ServerResourcesAccess	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/85ServerResourcesAccess	2006-11-01 20:52:28.000000000 -0500
@@ -0,0 +1,10 @@
+# Server resources access configuration
+
+<Directory /home/e-smith/files/server-resources>
+    Options +Indexes
+    order deny,allow
+    deny from all
+{
+    $OUT .= "    allow from $localAccess\n";
+}
+</Directory>
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess15brand mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess15brand
--- e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess15brand	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess15brand	2006-11-01 20:52:28.000000000 -0500
@@ -0,0 +1,12 @@
+#------------------------------------------------------------
+# e-smith files shared by manager and other control packages
+#------------------------------------------------------------
+
+<Directory "/home/e-smith/web/common">
+    Options Indexes Includes
+    AllowOverride None
+    order deny,allow
+    deny from all
+    allow from all
+</Directory>
+
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess15common mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess15common
--- e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess15common	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess15common	2006-11-01 20:52:28.000000000 -0500
@@ -0,0 +1,26 @@
+#------------------------------------------------------------
+# e-smith files shared by manager and other control packages
+#------------------------------------------------------------
+
+<Directory "/etc/e-smith/web/common">
+    Options Indexes Includes
+    AllowOverride None
+    order deny,allow
+    deny from all
+    allow from all
+</Directory>
+
+<Directory "/etc/e-smith/web/panels/manager/common">
+    Options Indexes Includes FollowSymLinks
+    AllowOverride None
+    order deny,allow
+    deny from all
+    allow from all
+    AuthName "Server manager"
+    AuthType basic
+    AuthExternal pwauth
+    require valid-user
+    SetEnv IMGHDR_SRC "/server-common/server-manager.jpg"
+    Satisfy all
+</Directory>
+
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess20manager mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess20manager
--- e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess20manager	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess20manager	2006-11-01 20:52:28.000000000 -0500
@@ -0,0 +1,32 @@
+
+#------------------------------------------------------------
+# e-smith-manager panel
+#------------------------------------------------------------
+
+<Directory "/etc/e-smith/web/panels/manager/html" >
+    Options Includes Indexes FollowSymLinks ExecCGI
+    AllowOverride None
+    order deny,allow
+    deny from all
+    allow from { $localAccess }
+    AuthName "Server manager"
+    AuthType Basic
+    AuthExternal pwauth
+    require user admin
+    SetEnv IMGHDR_SRC "/server-common/server-manager.jpg"
+    Satisfy all
+</Directory>
+
+<Directory "/etc/e-smith/web/panels/manager/cgi-bin">
+    Options Includes Indexes FollowSymLinks ExecCGI
+    AllowOverride None
+    order deny,allow
+    deny from all
+    allow from { $localAccess }
+    AuthName "Server manager"
+    AuthType Basic
+    AuthExternal pwauth
+    require user admin
+    SetEnv IMGHDR_SRC "/server-common/server-manager.jpg"
+    Satisfy all
+</Directory>
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess20password mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess20password
--- e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess20password	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/90e-smithAccess20password	2006-11-01 20:52:28.000000000 -0500
@@ -0,0 +1,18 @@
+
+#------------------------------------------------------------
+# e-smith-password panel
+#------------------------------------------------------------
+<Directory "/etc/e-smith/web/panels/password/html">
+    order deny,allow
+    deny from all
+    allow from { $localAccess }
+    SetEnv IMGHDR_SRC "/server-common/user-password.jpg"
+</Directory>
+
+<Directory "/etc/e-smith/web/panels/password/cgi-bin">
+    order deny,allow
+    deny from all
+    allow from { $localAccess }
+    SetEnv IMGHDR_SRC "/server-common/user-password.jpg"
+</Directory>
+
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/var/service/httpd-admin/log/run mezzanine_patched_e-smith-manager-1.13.0/root/var/service/httpd-admin/log/run
--- e-smith-manager-1.13.0/root/var/service/httpd-admin/log/run	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/var/service/httpd-admin/log/run	2006-11-01 20:54:46.000000000 -0500
@@ -0,0 +1,26 @@
+#!/bin/sh
+
+#----------------------------------------------------------------------
+# copyright (C) 2005 Mitel Networks Corporation
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA
+#
+# Technical support for this program is available from Mitel Networks
+# Please visit our web site www.mitel.com/sme/ for details.
+#----------------------------------------------------------------------
+exec                                    \
+    /usr/local/bin/setuidgid smelog     \
+    /usr/local/bin/multilog t s5000000  \
+    /var/log/httpd-admin
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.0/root/var/service/httpd-admin/run mezzanine_patched_e-smith-manager-1.13.0/root/var/service/httpd-admin/run
--- e-smith-manager-1.13.0/root/var/service/httpd-admin/run	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-manager-1.13.0/root/var/service/httpd-admin/run	2006-11-01 20:54:46.000000000 -0500
@@ -0,0 +1,11 @@
+#!/bin/sh
+#----------------------------------------------------------------------
+# copyright (C) 1999-2004 Mitel Networks Corporation
+#----------------------------------------------------------------------
+                                                                                    
+config=/etc/httpd/admin-conf/httpd.conf
+                                                                                    
+[ -e $config ] || exit 1
+                                                                                    
+exec 2>&1
+exec chpst -P /usr/sbin/httpd-admin -f $config -D FOREGROUND
