diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.1/root/etc/e-smith/templates/etc/e-smith/web/common/cgi-bin/AuthTKT.cfg/manager_access mezzanine_patched_e-smith-manager-1.13.1/root/etc/e-smith/templates/etc/e-smith/web/common/cgi-bin/AuthTKT.cfg/manager_access
--- e-smith-manager-1.13.1/root/etc/e-smith/templates/etc/e-smith/web/common/cgi-bin/AuthTKT.cfg/manager_access	2007-01-19 07:59:47.000000000 -0700
+++ mezzanine_patched_e-smith-manager-1.13.1/root/etc/e-smith/templates/etc/e-smith/web/common/cgi-bin/AuthTKT.cfg/manager_access	2007-01-19 07:53:38.000000000 -0700
@@ -1,5 +1,5 @@
 {
-    my $secret = ${httpd-admin}{TKTAuthSecret} || "34322500-7330-4400-423A-3A00434F5245";
+    my $secret = ${'httpd-admin'}{TKTAuthSecret} || "34322500-7330-4400-423A-3A00434F5245";
     $OUT .= "TKTAuthSecret $secret\n";
     $OUT .= "TKTAuthTimeoutMin 5";
 }
diff -Nur -x '*.orig' -x '*.rej' e-smith-manager-1.13.1/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20ManagerAuthTKT mezzanine_patched_e-smith-manager-1.13.1/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20ManagerAuthTKT
--- e-smith-manager-1.13.1/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20ManagerAuthTKT	2007-01-19 07:59:47.000000000 -0700
+++ mezzanine_patched_e-smith-manager-1.13.1/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20ManagerAuthTKT	2007-01-19 07:52:54.000000000 -0700
@@ -1,6 +1,6 @@
 {
     $OUT .= "LoadModule auth_tkt_module modules/mod_auth_tkt.so\n";
 
-    my $secret = ${httpd-admin}{TKTAuthSecret} || "34322500-7330-4400-423A-3A00434F5245";
+    my $secret = ${'httpd-admin'}{TKTAuthSecret} || "34322500-7330-4400-423A-3A00434F5245";
     $OUT .= "TKTAuthSecret \"$secret\"\n";
 }
